packages/protocol/test/L1/TaikoL1.integration.test.ts

```diff
@@ -0,0 +1,160 @@
+import { expect } from "chai";
+import { BigNumber, ethers as ethersLib } from "ethers";
+import { ethers } from "hardhat";
+import { TaikoL1, TaikoL2 } from "../../typechain";
+import { BlockMetadata } from "../utils/block_metadata";
+import { commitBlock, generateCommitHash } from "../utils/commit";
+import { buildProposeBlockInputs, proposeBlock } from "../utils/propose";
+import { deployTaikoL1 } from "../utils/taikoL1";
+import { deployTaikoL2 } from "../utils/taikoL2";
+
+describe("integration:TaikoL1", function () {
+    let taikoL1: TaikoL1;
+    let taikoL2: TaikoL2;
+    let l2Provider: ethersLib.providers.JsonRpcProvider;
+    let l2Signer: ethersLib.Signer;
+
+    beforeEach(async function () {
```

Comment:
shall we use fixture here? otherwise, will taikoL2 be deployed once per test case?

Reply:
i think we want it to be deployed once per test case? @davidtaikocha requested this change?

Reply:
if we need a fresh new contract for each testnet, i think `beforeEach` is more convenient? otherwise we can just use fixture.

Reply:
I think we should adopt one pattern as much as we can.